src/app/error.rs

```diff
@@ -7,3 +11,13 @@ pub enum Error {
     #[error("validation error: `{0}`")]
     Validation(String),
 }
+
+impl IntoResponse for Error {
+    fn into_response(self) -> Response {
+        match self {
+            Error::Database(e) => (StatusCode::INTERNAL_SERVER_ERROR, e.to_string()),
+            Error::Validation(e) => (StatusCode::BAD_REQUEST, e),
```

Comment:
Why DatabaseError calls to_string and ValidationError does not?

```rust 
 (StatusCode::INTERNAL_SERVER_ERROR, e.to_string())
```

```rust 
 (StatusCode::BAD_REQUEST, e)
```

Is there an especific reason?

Reply:
The return type of the match should be `(StatusCode, String)`. In that case, the enum's variants are `Error::Database(sqlx::error::Error)` and `Error::Validation(String)`. For the first one, we need to convert `sqlx::error::Error` to `String`. The other already has a `String` value.

Reply:
I got it. That's nice.